tests/e2e/virt_backup_restore_suite_test.go

```diff
@@ -53,47 +68,103 @@ func runVmBackupAndRestore(brCase VmBackupRestoreCase, expectedErr error, update
-	// DataVolume must be removed before backing up, because otherwise the
-	// restore process might try to follow the instructions in the spec. For
-	// example, a DataVolume that lists a cloned PVC will get restored in the
-	// same state, and attempt to clone the PVC again after restore. The
-	// DataVolume also needs to be detached from the VM, so that the VM restore
-	// does not try to use the DataVolume that was deleted.
-	err = v.DetachPvc(brCase.Namespace, diskName, 2*time.Minute)
-	gomega.Expect(err).To(gomega.BeNil())
-	err = v.RemoveDataVolume(brCase.Namespace, diskName, 2*time.Minute)
-	gomega.Expect(err).To(gomega.BeNil())
+		// Remove the Data Volume, but keep the PVC attached to the VM. The
+		// DataVolume must be removed before backing up, because otherwise the
+		// restore process might try to follow the instructions in the spec. For
+		// example, a DataVolume that lists a cloned PVC will get restored in the
+		// same state, and attempt to clone the PVC again after restore. The
+		// DataVolume also needs to be detached from the VM, so that the VM restore
+		// does not try to use the DataVolume that was deleted.
+		err = v.DetachPvc(brCase.Namespace, diskName, 2*time.Minute)
+		gomega.Expect(err).To(gomega.BeNil())
+		err = v.RemoveDataVolume(brCase.Namespace, diskName, 2*time.Minute)
+		gomega.Expect(err).To(gomega.BeNil())
+
+	} else { // Fedora test
+		err = lib.InstallApplication(v.Client, brCase.Template)
+		if err != nil {
+			fmt.Printf("Failed to install VM template %s: %v", brCase.Template, err)
+		}
+		gomega.Expect(err).To(gomega.BeNil())
+
+		// Wait for VM to start, then give some time for cloud-init to run.
+		// Afterward, run through the standard application verification to make sure
+		// the application itself is working correctly.
+		err = wait.PollImmediate(10*time.Second, 10*time.Minute, func() (bool, error) {
+			unstructuredVm, err := v.Dynamic.Resource(virtualMachineGVK).Namespace(brCase.Namespace).Get(context.Background(), brCase.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					log.Printf("VM %s/%s does not exist yet.", brCase.Namespace, brCase.Name)
+					return false, nil
+				}
+				log.Printf("Error getting virtual machine %s/%s: %v", brCase.Namespace, brCase.Name, err)
+				return false, err
+			}
+			if unstructuredVm == nil {
+				return false, nil
+			}
+			status, ok, err := unstructured.NestedString(unstructuredVm.UnstructuredContent(), "status", "printableStatus")
+			if err != nil {
+				log.Printf("Error getting status from VM: %v", err)
+				return false, err
+			}
+			if !ok {
+				log.Printf("Could not find VM status in API result.")
+				return false, nil
+			}
+			log.Printf("Status for VM %s/%s: %s", brCase.Namespace, brCase.Name, status)
+			return status == "Running", nil
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	}
+
+	if brCase.InitDelay > 0*time.Second {
```

Comment:
can be a TODO comment for later if we need this fast: can not we check an object status or something to see if it finished?

Reply:
Sure, I put in a TODO. I haven't thought of a great way to do this yet though.